server/index.js

```diff
@@ -81,67 +84,123 @@ function renderDocumentWithData(doc, data, reactComponent){
 
             var place = geoJson[0];
-            var list = [{'index': 0, 'pheromon_id': place.properties.pheromon_id}]
-            withPlacesMeasurements(list)
-            .then(function(measures){
 
-                if(measures !== null && measures.length > 0){
-                    place["measurements"] = {latest: measures[0].latest, max: measures[0].max};
+            if (place.properties.pheromon_id){
+
+                
+                var list = [{'index': 0, 'pheromon_id': place.properties.pheromon_id}]
+                withPlacesMeasurements(list)
+                .then(function(measures){
+
+                    if(measures !== null && measures.length > 0){
+                        place["measurements"] = {latest: measures[0].latest, max: measures[0].max};
+                    }
+                    if(raw){
+                        res.setHeader('Content-Type', 'application/json');
+                        res.send(JSON.stringify(place));
+                    } else {
+                        layoutData.detailedObject = place;
+                        renderAndSend(req, res, layoutData, placeScreen);
+                    }
+                })
+                .catch(function(err){
+                    console.error(err);
+                    if(raw){
+                        res.setHeader('Content-Type', 'application/json');
+                        res.send(JSON.stringify(place));
+                    } else {
+                        layoutData.detailedObject = place;
+                        renderAndSend(req, res, layoutData, placeScreen);
+                    }
+                });
+            }
+            else {
+                if(raw){
+                    res.setHeader('Content-Type', 'application/json');
+                    res.send(JSON.stringify(place));
+                } else {
+                    layoutData.detailedObject = place;
+                    renderAndSend(req, res, layoutData, placeScreen);
                 }
-                res.setHeader('Content-Type', 'application/json');
-                res.send(JSON.stringify(place));
-            })
-            .catch(function(err){
-                console.error(err);
-                res.setHeader('Content-Type', 'application/json');
-                res.send(JSON.stringify(place));
-            });
+            }
+        })
+        .catch(function(error){
+            res.status(500).send('Couldn\'t get place from database');
+            console.log('error in GET /place/' + placeId, error);
         });
     })
     .catch(function(error){
-        res.status(500).send('Couldn\'t get place from database');
+        res.status(500).send('Couldn\'t get place and measurements from database');
         console.log('error in GET /place/' + placeId, error);
     });
    
+}
+
+app.get('/place/:placeId/', function(req, res){
+    getPlace(req, res, false)
 });
 
+app.get('/rawPlace/:placeId/', function(req, res){
+    getPlace(req, res, true)
+});
```

Comment:
As discussed, let's make a single `/place/:placeId/` route that branches against the `Accept` header `req.headers['accept']` which contains `text/html` when asked by the browser and `application/json` from XHRs (but we need to set the header explicitly in the client-side for this to work)